### PR TITLE
fix: #25102 cryo pod now uses health-analyzer system to update body state in UI

### PR DESCRIPTION
**Pull request by Fildrance**
## About the PR
Cryo pod now uses health-analyzer system for ui to monitor body inside. Fixed Issue #25102 

## Why / Balance
Cryo pod used health-analyzer ui but was not using its subsystem, so after live updates on patient health was introduced it was not working in the same way, which seems off.

## Technical details
Attached HealthCheckComponent to cryopod, filling its ScannedEntity on opening UI, detaching it on ejecting body. Hoping nothing will break due to it...

## Media
[cryo-now-updates-ui-automatically.webm](https://github.com/space-wizards/space-station-14/assets/14752842/41712e83-ffab-4074-8960-7813fc755efe)

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: now cryo pod health analyzer updates atomatically!

### CODE REVIEW

**Comment by Fildrance** [comment]
HealthAnalyzerSystem is currently going to send update for every cryopod that any person have used, until body is ejected. It sounds like an unwanted side-effect but i'm not sure how to address it. Options are:
1. extract nessesary code from HealthAnalyzerSystem to HealthAnalyzerReportingSystem and create separate events, changing behaviour for cryopod so it IS auto-updating but it stops sending updates for UI after all players closed its console in any way possible: closing window, getting too far away, getting power cut off
2. adjust HealthAnalyzerSystem so it watches amount of open UIs and stops sending health-updates as long as there is no one watching monitor.

**Comment by Fildrance** [comment]
> HealthAnalyzerSystem is currently going to send update for every cryopod that any person have used, until body is ejected. It sounds like an unwanted side-effect but i'm not sure how to address it. Options are:
> 
> 1. extract nessesary code from HealthAnalyzerSystem to HealthAnalyzerReportingSystem and create separate events, changing behaviour for cryopod so it IS auto-updating but it stops sending updates for UI after all players closed its console in any way possible: closing window, getting too far away, getting power cut off
> 2. adjust HealthAnalyzerSystem so it watches amount of open UIs and stops sending health-updates as long as there is no one watching monitor.

But then again, hand-held health-analyzer works in the same manner, sync-ing even if you are not watching it.

File: Content.Server/Medical/CryoPodSystem.cs [hunk 1]
```diff
@@ -282,5 +289,16 @@ private void OnGasAnalyzed(Entity<CryoPodComponent> entity, ref GasAnalyzerScanE
         args.GasMixtures = gasMixDict;
     }
 
+    private void OnEjected(EntityUid uid, CryoPodComponent component, ref EntRemovedFromContainerMessage args)
```
Comment (TheShuEd) [inline Content.Server/Medical/CryoPodSystem.cs:292]: use new `Entity<>`
```suggestion
    private void OnEjected(Entity<CryoPodComponent> cryoPod, ref EntRemovedFromContainerMessage args)
```

**Review by TheShuEd — CHANGES_REQUESTED** [review]

File: Content.Server/Medical/CryoPodSystem.cs (hunk 1 above)
Comment (Fildrance) [reply Content.Server/Medical/CryoPodSystem.cs]: fixed

**Review by TheShuEd — APPROVED** [review]

**Pull request merged**